extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/RuntimeBeanProcessor.java

```diff
@@ -48,7 +48,11 @@ public void write(String name, byte[] data) {
             //deterministic name
             //as we know the maps are sorted this will result in the same hash for the same bean
             String name = bean.type.replace(".", "_") + "_" + HashUtil.sha1(bean.qualifiers.toString());
-            map.put(name, bean.supplier);
+            if (bean.runtimeValue != null) {
+                map.put(name, template.createSupplier(bean.runtimeValue));
```

Comment:
Wondering why you don't create the supplier here directly?

Reply:
The map is passed as a param to a template method. And we cannot record a lambda instance or an instance of annonymous class.